### PR TITLE
Drop support for Python 3.6

### PR DESCRIPTION
**Pull request by jrbourbeau**
For additional context see the discussion over in dask/community#66. There's a corresponding PR for dropping Python 3.6 in Dask at https://github.com/dask/dask/pull/7006.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,8 +6,7 @@ dist: trusty
 
 env:
   matrix:
-    - PYTHON=3.6 TESTS=true COVERAGE=true PACKAGES="lz4" TORNADO=5 CRICK=true
-    - PYTHON=3.7 TESTS=true PACKAGES="python-snappy python-blosc" TORNADO=6
+    - PYTHON=3.7 TESTS=true COVERAGE=true PACKAGES="lz4 python-snappy python-blosc" CRICK=true TORNADO=5
```
Comment (crusaderky) [inline .travis.yml:9]: I think here you removed the test case for when snappy and blosc are not installed

**Review by crusaderky — CHANGES_REQUESTED** [review]
One small tweak to CI needed + merge from master

**Comment by jrbourbeau** [comment]
Thanks for reviewing @crusaderky, I'll update accordingly

**Comment by martindurant** [comment]
Perhaps it would be best to restart this PR from scratch?

**Comment by jakirkham** [comment]
There's a Python 3.9 PR ( https://github.com/dask/distributed/pull/4460 ). Maybe we can just drop Python 3.6 in that? Or shortly thereafter?

**Comment by crusaderky** [comment]
> There's a Python 3.9 PR ( #4460 ). Maybe we can just drop Python 3.6 in that? Or shortly thereafter?

the plan is to drop 3.6 immediately after the 3.9 PR is merged

**Comment by jakirkham** [comment]
Would it makes sense to do this after the upcoming release? Sorry if this was already answered elsewhere

**Comment by jrbourbeau** [comment]
xref-ing https://github.com/dask/dask/pull/7006#issuecomment-791036320 from over in the corresponding PR in Dask. For Distributed dropping Python 3.6 in the next release has the (potential) benefit of resolving https://github.com/dask/distributed/issues/4538 (assuming it's only a rare Python 3.6 effect -- though this hasn't been completely confirmed)

**Comment by jakirkham** [comment]
Cool. Let's keep the discussion there. Then we can handle both PRs based on that outcome 🙂

File: .travis.yml (hunk 1 above)
Comment (jakirkham) [reply .travis.yml]: I think the new Python 3.7 environment file is not including snappy or blosc, but please recheck and make sure this looks ok 🙂

**Comment by jakirkham** [comment]
Looks like we have a conflict here. Would you be able to look at that James? 🙂

**Comment by jakirkham** [comment]
Oops that didn’t work as expected. Well if people here can take another look that would be appreciated 🙂

**Comment by crusaderky** [comment]
In metrics.py, you should drop ``_linux_thread_time()`` and ``_native_clock_func()`` and simplify their wrappers

**Review by crusaderky — APPROVED** [review]

**Review by jacobtomlinson — APPROVED** [review]

**Comment by jrbourbeau** [comment]
Thanks all for reviewing!

**Comment by jakirkham** [comment]
Thanks for working on this James! 😄

**Pull request merged**